x-pack/qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TokenBackwardsCompatibilityIT.java

```diff
@@ -14,197 +14,365 @@
-                "{\n" +
-                "    \"username\": \"test_user\",\n" +
-                "    \"password\": \"x-pack-test-password\",\n" +
-                "    \"grant_type\": \"password\"\n" +
-                "}");
-        Response response = client().performRequest(createTokenRequest);
-        assertOK(response);
-        Map<String, Object> responseMap = entityAsMap(response);
-        String token = (String) responseMap.get("access_token");
-        assertNotNull(token);
-        assertTokenWorks(token);
+        // Creates two access and refresh tokens and stores them in the token_backwards_compatibility_it index to be used for tests in the
+        // mixed/upgraded clusters
+        Map<String, Object> responseMap = createTokens(client(), "test_user", "x-pack-test-password");
+        String accessToken = (String) responseMap.get("access_token");
+        assertNotNull(accessToken);
+        assertAccessTokenWorks(accessToken);
+        String refreshToken = (String) responseMap.get("refresh_token");
+        assertNotNull(refreshToken);
 
-        Request indexRequest1 = new Request("PUT", "token_backwards_compatibility_it/_doc/old_cluster_token1");
-        indexRequest1.setJsonEntity(
-                "{\n" +
-                "    \"token\": \"" + token + "\"\n" +
-                "}");
-        Response indexResponse1 = client().performRequest(indexRequest1);
-        assertOK(indexResponse1);
-        Request createSecondTokenRequest = new Request("POST", "/_security/oauth2/token");
-        createSecondTokenRequest.setEntity(createTokenRequest.getEntity());
-        response = client().performRequest(createSecondTokenRequest);
-        responseMap = entityAsMap(response);
-        token = (String) responseMap.get("access_token");
-        assertNotNull(token);
-        assertTokenWorks(token);
-        Request indexRequest2 = new Request("PUT", "token_backwards_compatibility_it/_doc/old_cluster_token2");
-        indexRequest2.setJsonEntity(
-                "{\n" +
-                "    \"token\": \"" + token + "\"\n" +
-                "}");
-        Response indexResponse2 = client().performRequest(indexRequest2);
-        assertOK(indexResponse2);
+        storeTokens(client(), 1, accessToken, refreshToken);
+
+        responseMap = createTokens(client(), "test_user", "x-pack-test-password");
+        accessToken = (String) responseMap.get("access_token");
+        assertNotNull(accessToken);
+        assertAccessTokenWorks(accessToken);
+        refreshToken = (String) responseMap.get("refresh_token");
+        assertNotNull(refreshToken);
+
+        storeTokens(client(), 2, accessToken, refreshToken);
+    }
+
+    public void testRefreshingTokensInOldCluster() throws Exception {
+        assumeTrue("this test should only run against the old cluster", CLUSTER_TYPE == ClusterType.OLD);
+        // Creates another access and refresh tokens and tries to use the refresh tokens several times
```

Comment:
```suggestion
        // Creates another access and refresh token and tries to use the refresh tokens several times
```

Reply:
hmm, I would say that since there's two of them, the _s_ is correct?

Reply:
I _think_ what sounded strange was the combination of another + plural , but I'm not entirely sure

Reply:
hah, yeah, I'll remove _another_